build.gradle.kts

```diff
@@ -80,6 +80,8 @@ dependencies {
 
     // Testing
     testImplementation(libs.bundles.junit)
+    testImplementation(platform(libs.mockito.bom))
```

Comment:
I do not like mockito tests or want to add it. They are unintuitive (to me at least) and unreliable across changes long term. In this case you can just use a test player to get the packets (or write a little utility to do that and ensure none received if you dont like the boilerplate)

Reply:
From my point of view that would be a bit too much just “a bit” of quality for sure. What you suggest would be an integration test, during which I try to write an implementation test. Unfortunately, there is no better way in the current framework to catch such cases!